src/GivingCircle.Api.DataAccess/Repositories/IUserRepository.cs

```diff
@@ -0,0 +1,49 @@
+﻿using GivingCircle.Api.DataAccess.Responses;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace GivingCircle.Api.DataAccess.Repositories
+{
+    public interface IUserRepository
+    {
+        /// <summary>
+        /// Gets a user by their id
+        /// </summary>
+        /// <param name="userId">The users id</param>
+        /// <returns>the user object</returns>
+        Task<GetUserResponse> GetUserAsync(string userId);
+
+        /// <summary>
+        /// Creates a user
+        /// </summary>
+        /// <param name="user">The user</param>
+        /// <returns>True if success, false or an error if failure</returns>
+        Task<bool> CreateUserAsync(Models.User user);
+
+
+        /// <summary>
+        /// Updates the user using the given object
+        /// </summary>
+        /// <param name="userId">The user's id</param>
+        /// <param name="user">The user to update</param>
+        /// <returns>True if success, false or an error if un successful</returns>
+        Task<bool> UpdateUserAsync(string userId, Models.User user);
+
+        /// <summary>
+        /// Permanently deletes a user
+        /// </summary>
+        /// <param name="userId">The user's id</param>
+        /// <returns>True if success, else false if an error</returns>
+        Task<bool> DeleteUserAsync(string userId);
+
+        /// <summary>
+        /// Validates User
+        /// </summary>
+        /// <param name="userId">The users id</param>
+        /// <param name="password"> The users password</param>
+        /// <returns>True if validated, else false</returns>
+        Task<bool> ValidateUserIdAsync(string userId, string password);
```

Comment:
The user won't know their user id and to they'll have to login with their username and email or password, so this should return the user id to the front end instead of bool.

Reply:
This wasnt made for a login

Reply:
I was imagining this was assisting a separate controller that handled the login. We can make this for the login.